### PR TITLE
feat: add snippetgen config language and golden files

### PR DESCRIPTION
**Pull request by dizcology**
Prototype code for configurable snippet generation code.  This PR includes the snippet configuration language definition and golden test files only.

### CODE REVIEW

**Review by parthea — APPROVED** [review]
LGTM, please can you update the docs or include a README with more information about how these files were created?

File: tests/configurable_snippetgen/test_resources.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from gapic import utils
+from gapic.schema import api
+
+
+CURRENT_DIRECTORY = Path(__file__).parent.absolute()
+
+# This is a copy of the CodeGeneratorRequest dumped with:
+# protoc google/cloud/speech/v1/*.proto \
+# --experimental_allow_proto3_optional \
+# --proto_path=../api-common-protos/ --proto_path=. \
```
[the remark this answers is not in the document]
Comment (dizcology) [reply tests/configurable_snippetgen/test_resources.py:30]: Added `resources/README.md` for how to regenerate the file.  Also replaced `request.desc` with one dumped without using `api-common-protos`.

[the remark this answers is not in the document]
Comment (parthea) [reply tests/configurable_snippetgen/test_resources.py:30]: Please update the comment here also which mentions `api-common-protos`

**Review by parthea — APPROVED** [review]

File: tests/configurable_snippetgen/test_resources.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dizcology) [reply tests/configurable_snippetgen/test_resources.py]: Done - thanks!

**Pull request merged**